app/models/discount.rb

```diff
@@ -1,3 +1,7 @@
 class Discount < ApplicationRecord
-  belongs_to :order, optional: true
+  has_many :order, dependent: :destroy
+  scope :filter_discount_by_code, ->(code){where(code: code)}
```

Comment:
đặt tên scope dài quá, ở trong model Discount thì chắc chắn là filter discount rồi nên chỉ cần đặt là 
```suggestion
  scope :by_code, ->(code){where code: code if code.present?}
```
khi gọi sẽ là Discount.by_code thì clear hơn